File: web-service/src/main/java/Post/Poll.java
package Post;

import Notification.Notification;
import core.Permission;

/**
 *
 * @author marcw
 */
public class Poll extends AbstractPost {

	boolean isAnonymous;

	PollQuestion[] questions;

	@Override
	protected Notification generateNotification(String title, String message) {
		throw new UnsupportedOperationException("Not supported yet."); //To change body of generated methods, choose Tools | Templates.
	}

	boolean isClosed() {
		throw new UnsupportedOperationException("Not supported yet."); //To change body of generated methods, choose Tools | Templates.
	}

	@Override
	public Permission hasPermission() {
		throw new UnsupportedOperationException("Not supported yet."); //To change body of generated methods, choose Tools | Templates.
	}

	@Override
	public Object[] search(String query) {
		throw new UnsupportedOperationException("Not supported yet."); //To change body of generated methods, choose Tools | Templates.
	}
}

File: web-service/src/models/post/Poll.ts

/**
 * 
 * @author marcw
 */
export class Poll extends AbstractPost {
	isAnonymous: boolean;

	questions: PollQuestion[];

	generateNotification(title: string, message: string): Notification {
		return null;
	}

	isClosed(): boolean {
		return null;
	}

	public hasPermission(): Permission {
		return null;
	}

	public search(query: string): any[] {
		return null;
	}

	constructor() {
		super();
		this.isAnonymous = false;
	}
}
